reference/reference.go

```diff
@@ -110,6 +110,12 @@ func (r Spec) Hostname() string {
 	return r.Locator[:i]
 }
 
+// Path returns the path portion of the locator without the head slash.
+// e.g. "foo/bar" for "host/foo/bar:baz".
+func (r Spec) Path() string {
+	return strings.TrimPrefix(r.Locator, r.Hostname()+"/")
```

Comment:
This should use the URL parser to set extract this path.

Either way, I am not sure this should be an exported method on the spec.